src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -424,6 +424,27 @@ def subvolume_getpath(self, fs_handle, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    @connection_pool_wrap
+    def list_subvolumes(self, fs_handle, **kwargs):
+        ret        = 0, "", ""
+        groupname  = kwargs['group_name']
+
+        try:
+            with SubVolume(self.mgr, fs_handle) as sv:
+                spec = SubvolumeSpec(None, groupname)
+                if not self.group_exists(sv, spec):
+                    raise VolumeException(
+                        -errno.ENOENT, "Subvolume group '{0}' not found".format(groupname))
+                path = sv.get_group_path(spec)
+                if path is None:
+                    raise VolumeException(-errno.ENOENT,
+                                          "Subvolume group path '{0}' not found".format(spec.group_path))
```

Comment:
The error message doesn't seem very useful.

Since we don't explicitly create default subvolume group (it's created only after a subvolume is created without specifying a group name) and assume it always exists, I think we should just return *empty list* when we list subvolumes in the default subvolume group and no subvolume or default subvolume group is created.

Add a code comment saying why you're returning an empty list.
```
# When default subvolume group is not yet created we just return an empty list.
if path is None:
        return 0, '[]', ""
```

Reply:
Yes, that is the exact case here. `/volumes/_nogroup` is created by the `fs subvolume create`, but the user can run the `fs subvolume ls` before the `/volumes/_nogroup` creation.